tests/std/tests/P2321R2_views_zip/test.cpp

```diff
@@ -0,0 +1,838 @@
+                std::common_type_t<decltype(std::ranges::size(std::declval<AllView<RangeTypes>>()))...>>;
+            std::same_as<expected_size_type> auto zip_size = zipped_range.size();
+
+            assert(zip_size == std::ranges::size(tuple_element_arr));
+            STATIC_ASSERT(noexcept(zipped_range.size())
+                          == noexcept(std::ranges::min(
+                              {static_cast<expected_size_type>(std::declval<AllView<RangeTypes>>().size())...})));
+        }
+
+        STATIC_ASSERT(CanMemberSize<const ZipType> == (std::ranges::sized_range<const AllView<RangeTypes>> && ...));
+        if constexpr (CanMemberSize<const ZipType>) {
+            using expected_size_type = std::_Make_unsigned_like_t<
+                std::common_type_t<decltype(std::ranges::size(std::declval<const AllView<RangeTypes>>()))...>>;
+            std::same_as<expected_size_type> auto zip_size = zipped_range.size();
+
+            assert(zip_size == std::ranges::size(tuple_element_arr));
+            STATIC_ASSERT(noexcept(std::as_const(zipped_range).size())
+                          == noexcept(std::ranges::min(
+                              {static_cast<expected_size_type>(std::declval<const AllView<RangeTypes>>().size())...})));
+        }
+
+        const bool is_empty = std::ranges::empty(tuple_element_arr);
+
+        // Validate view_interface::empty() and view_interface::operator bool()
+        //
+        // From here on out, we'll be re-using concepts which we already verified to reduce
+        // redundancy.
+        STATIC_ASSERT(
+            CanMemberEmpty<ZipType> == (std::ranges::sized_range<ZipType> || std::ranges::forward_range<ZipType>) );
+        if constexpr (CanMemberEmpty<ZipType>) {
+            assert(zipped_range.empty() == is_empty);
+        }
+
+        // clang-format off
+        STATIC_ASSERT(
+            CanMemberEmpty<
+            const ZipType> == (std::ranges::sized_range<const ZipType> ||
+                std::ranges::forward_range<const ZipType>));
+        // clang-format on
+        if constexpr (CanMemberEmpty<const ZipType>) {
+            assert(std::as_const(zipped_range).empty() == is_empty);
+        }
+
+        STATIC_ASSERT(CanBool<ZipType> == CanMemberEmpty<ZipType>);
+        if constexpr (CanBool<ZipType>) {
+            assert(static_cast<bool>(zipped_range) != is_empty);
+        }
+
+        STATIC_ASSERT(CanBool<const ZipType> == CanMemberEmpty<const ZipType>);
+        if constexpr (CanBool<const ZipType>) {
+            assert(static_cast<bool>(std::as_const(zipped_range)) != is_empty);
+        }
+
+        // Validate contents of zipped range
+        assert(std::ranges::equal(zipped_range, tuple_element_arr, [](const auto& lhs_tuple, const auto& rhs_tuple) {
+            return do_tuples_reference_same_objects(lhs_tuple, rhs_tuple);
+        }));
+
+#pragma warning(push)
+#pragma warning(disable : 4127)

```

Comment:
Comment which warning you suppress
```suggestion
#pragma warning(disable : 4127) // Conditional Expression is Constant
```